hack/lib/util.sh

```diff
@@ -713,7 +727,7 @@ function kube::util::ensure-cfssl {
   fi
 
   mkdir -p "${cfssldir}"
-  pushd "${cfssldir}" > /dev/null
+  pushd "${cfssldir}" > /dev/null || exit 1
```

Comment:
Isn't this redundant?

Reply:
Hmm, why? The idea is that we exit if the change to the new directory fails, so that we don't do the operations in the current dir.

Reply:
a) This is handled by set -o errexit
b) I would expect this to return 1 rather than exit 1

Reply:
a) This is bit tricky because `util.sh` is meant to be used from other files, and it doesn't set the `errexit` option itself. Even though this function appears to be only called from files which have the option set, shellcheck is right when it complains that proper error handling is missing, since there is no guarantee that it's called from files which handle the errors right.
b) In the other places in the same function the errors are handled with `exit 1` and `exit 2`. I think we should follow the pattern and keep the exit calls in place.

However, this said, if you really would like the exit commands removed then I'll do so of course. :-)

Reply:
I do not think this is a good pattern:
* It is not applied consistently:
  - above we do `kube::util::ensure-temp-dir` not `kube::util::ensure-temp-dir || exit 1`.
* It would be annoying/madness to apply it consistently and append `|| exit 1` to every line
* Thus we should assume that errexit is set, and
  - edge cases where we do not want a failed command to blow up the script we should add an `|| true` or `if ! whatever; then something; fi` etc.

Reply:
Thanks for the comments! After some thought I think you are absolutely right. Having the particular checks with `|| return 1` will mean that they are handled as before in the `set -o errexit` scripts (since the function call will fail) and also potential other scripts have the possibility of dealing with the error as they see best. I made the fix as you suggested.

Reply:
return is definitely better than exit, but better would just be having neither, and using `|| true` in the rare case you want to ignore an error

Reply:
The shellcheck issue in question is https://github.com/koalaman/shellcheck/wiki/SC2164. If we don't want to have the `|| return 1` pattern added to directory changing, we can also add just the shellcheck directives disabling the warning for known-to-be-safe `pushd` and `popd` usage.

I think shellcheck identifies the failure to change directory such a major issue that it requires the error to be handled.

Also, if we really believe that all callers of the functions in `util.sh` should have `set -o errexit`, we might as well have set it it at the top of `util.sh`. This will make the shellcheck stop warning about the `cd` commands because the errors are known to be handled. This would be a major semantic change though and probably should be done outside of this PR.

Reply:
Thanks for the clarification!